crates/db_schema/src/utils.rs

```diff
@@ -153,6 +169,55 @@ async fn build_db_pool_settings_opt(settings: Option<&Settings>) -> Result<DbPoo
   Ok(pool)
 }
 
+fn establish_connection(config: &str) -> BoxFuture<ConnectionResult<AsyncPgConnection>> {
+  let fut = async {
+    let rustls_config = rustls::ClientConfig::builder()
+      .with_safe_defaults()
+      .with_root_certificates(root_certs())
+      .with_no_client_auth();
+    let tls = tokio_postgres_rustls::MakeRustlsConnect::new(rustls_config);
+    let (client, conn) = tokio_postgres::connect(config, tls)
+      .await
+      .map_err(|e| ConnectionError::BadConnection(e.to_string()))?;
+    tokio::spawn(async move {
+      if let Err(e) = conn.await {
+        error!("Database connection failed: {e}");
+      }
+    });
+    AsyncPgConnection::try_from(client).await
+  };
+  fut.boxed()
+}
+
+fn root_certs() -> rustls::RootCertStore {
+  let mut roots = rustls::RootCertStore::empty();
+  let certs = load_certs_from_env()
+    .unwrap_or_else(rustls_native_certs::load_native_certs)
+    .expect("Certs not loadable!");
+  let certs: Vec<_> = certs.into_iter().map(|cert| cert.0).collect();
+  roots.add_parsable_certificates(&certs);
```

Comment:
Why is so much custom code necessary here? Is there a PR or issue for diesel-async to support this natively?

Reply:
Dang that's unfortunate. Oh well, as long as this is confined to its own functions, I spose its fine.

Reply:
Added a comment explaining that diesel-async does not support TLS connections out of the box